libs/datamodel/core/src/transform/ast_to_dml/db/walkers.rs

```diff
@@ -129,14 +137,45 @@ impl<'ast, 'db> ModelWalker<'ast, 'db> {
             })
     }
 
-    pub(crate) fn walk_relation_field(&self, field_id: FieldId) -> RelationFieldWalker<'ast, 'db> {
+    pub(crate) fn relation_field(&self, field_id: ast::FieldId) -> RelationFieldWalker<'ast, 'db> {
         RelationFieldWalker {
             model_id: self.model_id,
             field_id,
             db: self.db,
             relation_field: &self.db.types.relation_fields[&(self.model_id, field_id)],
         }
     }
+
+    pub(super) fn explicit_forward_relations(&self) -> impl Iterator<Item = ExplicitRelationWalker<'ast, 'db>> + '_ {
```

Comment:
I think the walker and the method are not named as they should, they consider only _complete_ relations, that's not all explicit relations.